client/allochealth/tracker.go

```diff
@@ -278,12 +278,15 @@ func (t *Tracker) watchTaskEvents() {
 		latestStartTime := time.Time{}
 		for taskName, state := range alloc.TaskStates {
 			// One of the tasks has failed so we can exit watching
-			if state.Failed || (!state.FinishedAt.IsZero() && !t.lifecycleTasks[taskName]) {
+			if state.Failed || (!state.FinishedAt.IsZero() && (t.lifecycleTasks[taskName]) == "") {
```

Comment:
Can we split the lifecycle check into it's own `if ... { continue }` with a comment?

Reply:
yep!